services/hotelsService.js

```diff
@@ -1,59 +1,64 @@
 const { Hotel } = require('../models/hotel')
-const Reservation = require('../models/reservation')
+const isRoomAvailable = require('../helpers/isRoomAvailable')
 
 const { BadRequestError, NotFoundError } = require('../helpers/apiError')
 const { formatDate } = require('../helpers/date')
 
 const DEFAULT_PAGE_SIZE = 50
 
-exports.getFreeRooms = async (req) => {
-  if (!req.query.startDate || !req.query.endDate)
-    throw new BadRequestError('Provide start date and end date.')
+const isHotelAvailable = async (hotelId, startDate, endDate, adults, children) => {
+  const hotel = await Hotel.findOne({ _id: hotelId })
 
-  const { id: hotelId } = req.params
-
-  let { startDate, endDate } = req.query
-
-  startDate = formatDate(startDate, true)
-  endDate = formatDate(endDate, true)
+  if(!hotel) {
+    throw new NotFoundError('Hotel not found.')
+  }
 
-  const hotel = await Hotel.findById(hotelId)
-  if (!hotel) throw new BadRequestError('Hotel not found')
+  const rooms = hotel.rooms.filter(r => r.beds.single + r.beds.double * 2 >= parseInt(adults) + parseInt(children))
 
-  const freeRooms = []
-  const rooms = hotel.rooms
+  if(rooms.length == 0) return false
 
-  for (const room of rooms) {
-    const roomReservations = await Reservation.find({ room: room._id })
-    if (roomReservations.length == 0) {
-      freeRooms.push(room)
-    } else {
-      const occupiedRR = roomReservations.some(
-        (rr) =>
-          (rr.startDate.toISOString() <= startDate &&
-            rr.endDate.toISOString() > startDate) ||
-          (rr.startDate.toISOString() < endDate &&
-            rr.endDate.toISOString() >= endDate)
-      )
-      if (!occupiedRR) freeRooms.push(room)
-    }
+  let isAvailable = false
+  for(r of rooms) {
+    isAvailable = await isRoomAvailable(hotelId, r._id, formatDate(startDate, true), endDate)
+    if(isAvailable) break 
   }
-  return freeRooms
+
+  return isAvailable
 }
 
 exports.getHotels = async (req) => {
-  const { city } = req.query
+  const { city, isFree, adults, children, startDate, endDate } = req.query
```

Comment:
what for 'isFree' is?

Reply:
It tells us if we want to get filtered hotels by selected filtration

Reply:
I think is not needed. City is already used without checking it. You can just check if startDate and endDate are provided or not

Reply:
Yeah, i see it now